functions/supplier/file-supplier/src/main/java/org/springframework/cloud/fn/supplier/file/FileSupplierConfiguration.java

```diff
@@ -93,10 +95,17 @@ else if (this.fileSupplierProperties.getFilenameRegex() != null) {
 	}
 
 	@Bean
-	@CustomizationAware
-	public FileInboundChannelAdapterSpec fileMessageSource(FileListFilter<File> fileListFilter) {
-		return Files.inboundAdapter(this.fileSupplierProperties.getDirectory())
+	@CustomizationDoc
+	public FileInboundChannelAdapterSpec fileMessageSource(FileListFilter<File> fileListFilter,
```

Comment:
I would bite a bullet and go ahead with this breaking change.
I believe we have done similar in the past without not objections from end-users.

It is really hard to imagine that someone extends our `@Configuration` classes for their purposes...

But if you insist for consistency, then indeed such a change should go only to the next version.

We have now an auto-configuration committed which provides a already a smooth end-user experience.
And yes: that comes without breaking changes in the target configurations at all...

I know that we try to make this a extendable library, but so far we deal only with closed box components to be available in the final just via auto-configuration.

So, probably better to have a consolidated team decision without my personal preferences.

Reply:
I vote for the approach outlined in this proposal rather than the previous auto-configuration based approach. 

I value the simplicity of this approach over the the "risk" of breaking the bean method w/ extra param. It is a very minor break if in fact someone is using the bean method in that manner.